lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -439,5 +406,23 @@ private async Task UnpackArchiveAsync(string archivePath, string outputPath, str
                 }
             }
         }
+
+        private async Task DownloadFileUsingHttpClientTaskAsync(string address, string filename)
+        {
+            using var handler = new HttpClientHandler();
+            if (WebProxy != null)
+            {
+                handler.Proxy = WebProxy;
+                handler.UseProxy = true;
+            }
+
+            using var client = new HttpClient(handler);
+
+            // Send the GET request and retrieve the response as a byte array
+            var fileBytes = await client.GetByteArrayAsync(address).ConfigureAwait(false);
+
+            // Write the byte array to a file
+            File.WriteAllBytes(filename, fileBytes);
```

Comment:
To write the data directly to disk instead of having a complete copy in memory
```suggestion
            // Send the GET request and retrieve the response as a stream
            using var downloadStream = await client.GetStreamAsync(address).ConfigureAwait(false);

            // Write the stream to a file
            using var fileStream = File.Create(filename);
            await downloadStream.CopyToAsync(fileStream).ConfigureAwait(false);
```

Reply:
Thank you!